spectractor/extractor/spectrum.py

```diff
@@ -654,107 +749,204 @@ def load_spectrum(self, input_file_name, spectrogram_file_name_override=None,
-                    self.load_spectrogram(spectrogram_file_name)
-                else:
-                    raise FileNotFoundError(f"No spectrogram info in {input_file_name} "
-                                            f"and not even a spectrogram file {spectrogram_file_name}.")
-                if "PSF_TAB" in hdu_list:
-                    self.chromatic_psf.init_table(Table.read(hdu_list["PSF_TAB"]),
-                                                  saturation=self.spectrogram_saturation)
-                elif os.path.isfile(psf_file_name):  # retro-compatibility
-                    self.my_logger.info(f'\n\tLoading PSF from {psf_file_name}...')
-                    self.load_chromatic_psf(psf_file_name)
+    def load_spectrum_latest(self, input_file_name):
+        """Load the spectrum from a fits file (data, error and wavelengths).
+
+        Parameters
+        ----------
+        input_file_name: str
+            Path to the input fits file
+
+        Examples
+        --------
+        >>> s = Spectrum(config="")
+        >>> s.load_spectrum('tests/data/reduc_20170530_134_spectrum.fits')
+        >>> print(s.units)
+        erg/s/cm$^2$/nm
+
+        .. doctest::
+            :hide:
+
+            >>> assert parameters.OBS_CAMERA_ROTATION == s.header["CAM_ROT"]
+            >>> assert parameters.CCD_REBIN == s.header["REBIN"]
+            >>> assert parameters.OBS_OBJECT_TYPE == "STAR"
+            >>> assert s.parallactic_angle == s.header["PARANGLE"]
+
+        """
+        self.header, raw_data = load_fits(input_file_name)
+        self.lambdas = raw_data[0]
+        self.lambdas_binwidths = np.gradient(self.lambdas)
+        self.data = raw_data[1]
+        if len(raw_data) > 2:
+            self.err = raw_data[2]
+            self.cov_matrix = np.diag(self.err ** 2)
+
+        # set the config parameters first
+        if self.config == "":
+            param_header, _ = load_fits(input_file_name, hdu_index="CONFIG")
+            for key in param_header.keys():
+                if "X__" not in key and (not isinstance(param_header[key], str) or (isinstance(param_header[key], str) and "X__" not in param_header[key])):
+                    setattr(parameters, key, param_header[key])
+                elif "X__" in key:
+                    continue
+                elif "X__" in param_header[key]:
+                    setattr(parameters, key, param_header[param_header[key]])
                 else:
-                    raise FileNotFoundError(f"No PSF info in {input_file_name} "
-                                            f"and not even a PSF file {psf_file_name}.")
-                if "LINES" in hdu_list:
-                    self.lines.table = Table.read(hdu_list["LINES"], unit_parse_strict="silent")
-                hdu_list.close()
-        else:
-            raise FileNotFoundError(f'\n\tSpectrum file {input_file_name} not found')
+                    continue
```

Comment:
I think if you rewrite this using the `.items()` it will be a lot more readable and the logic easier to follow:

```
for key, value in param_header.items():
    if "X__" not in key and (not isinstance(value, str) or (isinstance(value, str) and "X__" not in value)):
        setattr(parameters, key, value)
    elif "X__" in key:
        continue
    elif "X__" in value:
        setattr(parameters, key, param_header[value])
    else:
        continue
```

Reply:
I did not know, thanks !